### PR TITLE
Adding New Shadows

### PR DESCRIPTION
**Pull request by masterrr**
screenshot_2015-05-17-23-49-53
Issue: https://github.com/toggl/mobile/issues/628

### CODE REVIEW

**Comment by monday8am** [comment]
@masterrr thanks! I'm checking it.

**Comment by taavirehemagi** [comment]
screen shot 2015-05-18 at 11 11 17
Could we add shadows to project selection screen as well in this PR?

**Comment by taavirehemagi** [comment]
screen shot 2015-05-18 at 11 13 02
@pe0ny  I'm not sure, but maybe the shadow should go the other way here, the TE list being the upper element.

**Comment by monday8am** [comment]
@taavirehemagi yes we should, but in another issue ;)

**Comment by taavirehemagi** [comment]
alright. Other than  that, OK from me.

File: Joey/UI/Views/ShadowItemDecoration.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+            if (childCount == 0) {
+                return;
+            }
+
+            for (int i = 0; i+1 < childCount; i++) {
+                var child = parent.GetChildAt (i);
+                var childNext = parent.GetChildAt (i+1);
+
+                if (!ShouldDraw (child) || childNext.GetType () == typeof (T)) {
```
Comment (monday8am) [inline Joey/UI/Views/ShadowItemDecoration.cs]: Replace:
```childNext.GetType () == typeof (T)```
by
```childNext is T```

You can activate the source analysis in Xamarin.

File: Joey/Joey.csproj [hunk 2]
```diff
@@ -669,6 +670,7 @@
     <AndroidResource Include="Resources\layout\EditGroupedTimeEntryFragment.axml" />
     <AndroidResource Include="Resources\layout\EditGroupedTimeEntryItem.axml" />
     <AndroidResource Include="Resources\layout\TogglTagsField.axml" />
+    <AndroidResource Include="Resources\drawable\DropShadowNew.xml" />
```
Comment (monday8am) [inline Joey/Joey.csproj]: DropShadowNew.xml: I know this file is new, but in the next commit will be old. Change name to DropShadowVertical or something more descriptive.

**Comment by monday8am** [comment]
@masterrr : following the material design rules, the shadow on items should be ordered by elevations. More elevation means bigger shadow. Toolbar should have a shadow bigger than items:
-- header elevation: 0
-- items elevation: 2 or 3
-- toolbar elevation: 4 or 6

We are not using elevation but we should accomplish a similar effect.

**Comment by masterrr** [comment]
@monday8am ready to be reviewed

**Comment by masterrr** [comment]
screenshot_2015-05-18-23-06-47

**Comment by masterrr** [comment]
@monday8am fixed

**Comment by monday8am** [comment]
@masterrr Item shadows (14dp) should be shorter than Toolbar shadow (3dp), like real world. 

```private const int shadowHeight = 14;```

**Comment by masterrr** [comment]
@monday8am I did like in @pe0ny's stuff from there https://github.com/toggl/mobile/issues/628, the bottom item's shadow is clearly bigger than a toolbar one

**Comment by monday8am** [comment]
@masterrr I understand, if you used the design and not the Google docs and your logic, you're doing wrong again. Please, read this documents:

https://developer.android.com/training/material/shadows-clipping.html
http://www.google.com/design/spec/what-is-material/objects-in-3d-space.html

use your logic and don't make me lose my time with this kind of answers.

**Comment by masterrr** [comment]
screenshot_2015-05-19-15-33-14

**Pull request merged**